src/app.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
-// Write your code here
+const readline = require('node:readline');
+const { stdin, stdout, exit } = require('node:process');
+
+const { generateRandomNumber } = require('./modules/generateRandomNumber');
+const { checkIsValidUserInput } = require('./modules/checkIsValidUserInput');
+const { getBullsAndCows } = require('./modules/getBullsAndCows');
+
+const PROMPT_MESSAGE = 'Write 4 digits: \n';
+const INVALID_INPUT_MESSAGE = 'The input must be 4 uniq digits: \n';
+const TRY_AGAIN_MESSAGE = 'Need to try one more time: \n';
+const WIN_MESSAGE = 'You win! The correct answer is';
+
+const input = readline.createInterface({ input: stdin, output: stdout });
+const answer = generateRandomNumber();
+
+stdout.write(PROMPT_MESSAGE);
+
+input.on('line', (text) => {
+  if (!checkIsValidUserInput(text)) {
+    stdout.write(INVALID_INPUT_MESSAGE);
+  }
```

Comment:
Currently, if the input is invalid, the program only writes an error message but does not return or prompt the user to input again. You should add a `return` statement after writing the `INVALID_INPUT_MESSAGE` to prevent further execution of the code block and prompt the user to try again.